RNA-seq/00c-tximeta_other_species.Rmd

```diff
@@ -0,0 +1,267 @@
+                release = "95", # Replace this with the release number 
+                genome = "GRCz11.103", # Replace with the genome version noted in the gtf file
+                fasta = fasta_urls,
+                gtf = gtf_url)
+```
+
+Once you have set up the `linkedTxome` for a species, you should not need to do it again (on a particular computer). 
+From this point on, we are doing the same `tximeta` steps we've used for human and mouse. 
+
+# Running tximeta for a zebrafish dataset
+
+The above section only needs to be completed once per species-genome release combination.
+The following steps in this section perform the dataset specific `tximeta` steps in the same way we would for human and mouse data. 
+
+For this example, we are using a [zebrafish cortisol experiment dataset](https://www.refine.bio/experiments/SRP073217/response-to-low-dose-cortisol-treatment-in-zebrafish-embryos-larvae) that we have already quantified with salmon. 
+This means that the `quant.sf` files are ready for use with `tximeta()`. 
+
+![](diagrams/rna-seq_5.png)
+
+## Directories and file names
+
+These directories and files are set up so we can point to the `quant.sf` files. 
+
+```{r}
+# Directory where the data are located
+data_dir <- file.path("data", "zebrafish-cortisol")
+
+# Directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path(data_dir, "salmon_quant")
+
+# Create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+```{r input-names}
+# Identify the quant files themselves and store them in a vector, `sf_files`
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+```{r metadata-file}
+# sample metadata file
+meta_file <- file.path(data_dir, "zebrafish-cortisol_metadata.tsv")
+```
+
+**Output**
+
+```{r output-names}
+# Name the output zebrafish-cortisol_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "zebrafish-cortisol_tximeta.RDS")
+```
+
+## File names
+
+```{r sf_files}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` package.
+
+```{r sample_names}
+sample_names <- stringr::word(sf_files, -2, sep = "/")
+sample_names
+```
+
+## Set up metadata
+
+```{r names_sf_files}
+coldata <- data.frame(files = sf_files, 
+                      names = sample_names)
+```
+
+We have more information about these samples stored in the metadata file that we will also want stored in `coldata`. 
+Let's read in the sample metadata from the TSV file and do a bit of set up. 
+We really only want two columns from this metadata file, so we will use a `dplyr::select()` to keep only those. 
```

Comment:
I was trying to make sure the tximeta data output we made was still reasonably useable in case participants wanted to do some further analyses, but I didn't bother doing the splitting. 

It would be just one line. And now I'm thinking if a participant does use this data they are going to ask us how to do this anyway, so might as well do it now.